sw/host/opentitanlib/src/test_utils/lc_transition.rs

```diff
@@ -0,0 +1,146 @@
+    }
+
+    // Check the LC transition mutex has not been claimed yet.
+    if jtag.read_lc_ctrl_reg(&LcCtrlReg::ClaimTransitionIf)? == u8::from(MultiBitBool8::True) as u32
+    {
+        return Err(LcTransitionError::MutexAlreadyClaimed.into());
+    }
+
+    // Attempt to claim the LC transition mutex.
+    jtag.write_lc_ctrl_reg(
+        &LcCtrlReg::ClaimTransitionIf,
+        u8::from(MultiBitBool8::True) as u32,
+    )?;
+
+    // Check the LC transition mutex was claimed.
+    if jtag.read_lc_ctrl_reg(&LcCtrlReg::ClaimTransitionIf)? != u8::from(MultiBitBool8::True) as u32
+    {
+        return Err(LcTransitionError::FailedToClaimMutex.into());
+    }
+
+    // Program the target LC state, i.e., Scrap.
+    jtag.write_lc_ctrl_reg(
+        &LcCtrlReg::TransitionTarget,
+        target_lc_state.redundant_encoding(),
+    )?;
+
+    // Check correct target LC state was programmed.
+    let target_lc_state_programmed = jtag.read_lc_ctrl_reg(&LcCtrlReg::TransitionTarget)?;
+    if target_lc_state_programmed != target_lc_state.redundant_encoding() {
+        return Err(LcTransitionError::TargetProgrammingFailed(target_lc_state_programmed).into());
+    }
+
+    // Configure external clock.
+    if use_external_clk {
+        jtag.write_lc_ctrl_reg(
+            &LcCtrlReg::TransitionCtrl,
+            LcCtrlTransitionCtrlBit::union([LcCtrlTransitionCtrlBit::ExtClockEn]),
+        )?;
+    } else {
+        jtag.write_lc_ctrl_reg(&LcCtrlReg::TransitionCtrl, 0)?;
+    }
+
+    // Check external clock was not requested.
+    assert_ne!(
+        jtag.read_lc_ctrl_reg(&LcCtrlReg::TransitionCtrl)?,
+        LcCtrlTransitionCtrlBit::union([LcCtrlTransitionCtrlBit::ExtClockEn]),
+    );
+
+    // Initiate LC transition and poll status register until transition is completed.
+    jtag.write_lc_ctrl_reg(
+        &LcCtrlReg::TransitionCmd,
+        LcCtrlTransitionCmdBit::union([LcCtrlTransitionCmdBit::Start]),
+    )?;
+    loop {
+        let status = jtag.read_lc_ctrl_reg(&LcCtrlReg::Status)?;
+        if (status & LcCtrlStatusBit::union([LcCtrlStatusBit::TransitionSuccessful])) != 0 {
+            break;
+        }
+        if (status
+            & !LcCtrlStatusBit::union([
+                LcCtrlStatusBit::Initialized,
+                LcCtrlStatusBit::Ready,
+                LcCtrlStatusBit::TransitionSuccessful,
+            ]))
+            != 0
+        {
+            return Err(LcTransitionError::TransitionFailed.into());
+        }
+    }
+
+    // Check we have entered the post transition state.
+    let post_transition_lc_state = jtag.read_lc_ctrl_reg(&LcCtrlReg::LcState)?;
+    if post_transition_lc_state != DifLcCtrlState::PostTransition.redundant_encoding() {
+        return Err(LcTransitionError::BadPostTransitionState(post_transition_lc_state).into());
+    }
+
+    // Reset the chip, keeping LC TAP selected.
+    jtag.disconnect()?;
+    transport.reset_target(reset_delay, true)?;
+    jtag.connect(JtagTap::LcTap)?;
```

Comment:
Resetting within this function feels like unexpected behaviour, though I realise it's not that useful to have the device be put into a terminal state

Reply:
Disagreed, this is a test_util function, which is code designed to help us promote code reuse when writing tests. Performing an LC transition requires a reset, just like performing bootstrap does.